Errors while Linking 2D debug template.

### Tested versions

Current master.

### System information

Fedora

### Issue description

```
[ 75%] Linking Program bin/godot.linuxbsd.template_debug.x86_64 ...
[ 75%] /usr/bin/ld: bin/obj/scene/libscene.linuxbsd.template_debug.x86_64.a(navigation_mesh.linuxbsd.template_debug.x86_64.o): in function `NavigationMesh::get_debug_mesh()':
navigation_mesh.cpp:(.text+0x7be4): undefined reference to `NavigationServer3D::get_singleton()'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x7bec): undefined reference to `NavigationServer3D::get_debug_navigation_enable_geometry_face_random_color() const'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x7cbe): undefined reference to `NavigationServer3D::get_singleton()'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x7ccb): undefined reference to `NavigationServer3D::get_debug_navigation_geometry_face_material()'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x7d29): undefined reference to `NavigationServer3D::get_singleton()'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x7d31): undefined reference to `NavigationServer3D::get_debug_navigation_enable_edge_lines() const'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x80f3): undefined reference to `NavigationServer3D::get_singleton()'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x8100): undefined reference to `NavigationServer3D::get_debug_navigation_geometry_edge_material()'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x8473): undefined reference to `NavigationServer3D::get_singleton()'
/usr/bin/ld: navigation_mesh.cpp:(.text+0x8483): undefined reference to `NavigationServer3D::get_debug_navigation_geometry_face_color() const'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.linuxbsd.template_debug.x86_64] Error 1
scons: building terminated because of errors.
```

### Steps to reproduce

use `scons target=template_debug disable_3d=yes`

Regression from: https://github.com/godotengine/godot/pull/89409

### Minimal reproduction project (MRP)

N/A

--- a/servers/SCsub
+++ b/servers/SCsub
@@ -14,18 +14,11 @@ SConscript("display/SCsub")
 SConscript("movie_writer/SCsub")
 SConscript("rendering/SCsub")
 SConscript("text/SCsub")
-
-if not env["disable_navigation_2d"]:
-    SConscript("navigation_2d/SCsub")
-if not env["disable_physics_2d"]:
-    SConscript("physics_2d/SCsub")
-
-if not env["disable_navigation_3d"]:
-    SConscript("navigation_3d/SCsub")
-if not env["disable_physics_3d"]:
-    SConscript("physics_3d/SCsub")
-if not env["disable_xr"]:
-    SConscript("xr/SCsub")
+SConscript("navigation_2d/SCsub")
+SConscript("physics_2d/SCsub")
+SConscript("navigation_3d/SCsub")
+SConscript("physics_3d/SCsub")
+SConscript("xr/SCsub")
 
 lib = env.add_library("servers", env.servers_sources)
 

--- a/servers/navigation_2d/SCsub
+++ b/servers/navigation_2d/SCsub
@@ -3,4 +3,5 @@ from misc.utility.scons_hints import *
 
 Import("env")
 
-env.add_source_files(env.servers_sources, "*.cpp")
+if not env["disable_navigation_2d"]:
+    env.add_source_files(env.servers_sources, "*.cpp")

--- a/servers/navigation_3d/SCsub
+++ b/servers/navigation_3d/SCsub
@@ -3,4 +3,8 @@ from misc.utility.scons_hints import *
 
 Import("env")
 
-env.add_source_files(env.servers_sources, "*.cpp")
+if not env["disable_navigation_3d"]:
+    env.add_source_files(env.servers_sources, "*.cpp")
+else:
+    if env.debug_features:
+        env.add_source_files(env.servers_sources, "navigation_server_3d.cpp")

--- a/servers/physics_2d/SCsub
+++ b/servers/physics_2d/SCsub
@@ -3,4 +3,5 @@ from misc.utility.scons_hints import *
 
 Import("env")
 
-env.add_source_files(env.servers_sources, "*.cpp")
+if not env["disable_physics_2d"]:
+    env.add_source_files(env.servers_sources, "*.cpp")

--- a/servers/physics_3d/SCsub
+++ b/servers/physics_3d/SCsub
@@ -3,4 +3,5 @@ from misc.utility.scons_hints import *
 
 Import("env")
 
-env.add_source_files(env.servers_sources, "*.cpp")
+if not env["disable_physics_3d"]:
+    env.add_source_files(env.servers_sources, "*.cpp")

--- a/servers/xr/SCsub
+++ b/servers/xr/SCsub
@@ -3,4 +3,5 @@ from misc.utility.scons_hints import *
 
 Import("env")
 
-env.add_source_files(env.servers_sources, "*.cpp")
+if not env["disable_xr"]:
+    env.add_source_files(env.servers_sources, "*.cpp")

